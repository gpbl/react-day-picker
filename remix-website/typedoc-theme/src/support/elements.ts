/**
 * A set of pure functions that returns markdown elements as strings.
 *
 * @module
 */

import { formatContents, trimLastLine, unEscapeChars } from './utils.js';

export const heading = (level: number, text: string) => {
  level = level > 6 ? 6 : level;
  return `${[...Array(level)].map(() => '#').join('')} ${text}`;
};

export const link = (label: string, url: string | null) =>
  url ? `[${label}](${url})` : '';

export const bold = (text: string) => `**${text}**`;

export const italic = (text: string) => `*${text}*`;

export const backTicks = (text: string) =>
  /(\`)/g.test(text) ? text.replace(/`/g, '\\`') : `\`${text}\``;

export const unorderedList = <T>(items: T[]) =>
  items.map((item) => `- ${item}`).join('\n');

export const horizontalRule = () => '\n\n***\n\n';

export const codeBlock = (content: string) => {
  const trimmedContent =
    content.endsWith('}') ||
    content.endsWith('};') ||
    content.endsWith('>') ||
    content.endsWith('>;')
      ? trimLastLine(content)
      : content;
  return '```ts\n' + unEscapeChars(trimmedContent) + '\n```';
};

export const strikeThrough = (content: string) => `~~${content}~~`;

export const table = (headers: string[], rows: string[][]) =>
  `\n| ${headers.join(' | ')} |\n| ${headers
    .map(() => ':------')
    .join(' | ')} |\n${rows.map((row) => `| ${row.join(' | ')} |\n`).join('')}`;

export const blockQuoteBlock = (content: string) => {
  const lines = formatContents(content).split('\n');
  return lines
    .map((line) => (line.length ? `> ${line.trim()}` : '>'))
    .join('\n');
};

export const indentBlock = (content: string) => {
  const lines = content.split('\n');
  return lines
    .filter((line) => Boolean(line.length))
    .map((line) => `    ${line}`)
    .join('\n');
};
