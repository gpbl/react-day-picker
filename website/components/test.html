<!DOCTYPE html>
<html>
<head>
<style>
body {
  display: flex;
  margin: 0;
  padding: 0;
  min-height: 100vh;
  overflow-x: hidden;
}

.sidebar {
  position: sticky;
  top: 0;
  height: 100vh;
  width: 200px;
  background-color: #f5f5f5;
  padding: 20px;
  overflow-y: auto;
}

.main-content {
  flex-grow: 1;
  padding: 20px;
}
</style>
</head>
<body>

<div class="sidebar">
  <h2>Sidebar</h2>
  <p>This is a sticky sidebar.</p>
</div>

<div class="main-content">
  <h2>Main Content</h2>
  <p>Scroll the page.</p>
  <!-- Add enough content here so that the page scrolls -->
</div>

</body>
</html>
