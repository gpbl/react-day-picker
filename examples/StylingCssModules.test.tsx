import React from "react";

import { app } from "@/test/elements";
import { renderApp } from "@/test/renderApp";

import { CssModules } from "./CssModules";

const today = new Date(2021, 10, 25);
jest.useFakeTimers().setSystemTime(today);

beforeEach(() => {
  renderApp(<CssModules />);
});

test("should match the snapshot", () => {
  expect(app()).toMatchSnapshot();
});
