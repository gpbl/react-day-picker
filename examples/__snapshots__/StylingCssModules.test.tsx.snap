// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  role="application"
>
  <div
    class="rdp-root"
    data-mode="single"
  >
    <div
      class="rdp-months"
    >
      <nav
        aria-label=""
        class="rdp-nav"
      >
        <button
          aria-label="Go to the Previous Month"
          class="rdp-button_previous"
          type="button"
        >
          <svg
            class="rdp-chevron"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <polygon
              points="16 18.112 9.81111111 12 16 5.87733333 14.0888889 4 6 12 14.0888889 20"
            />
          </svg>
        </button>
        <button
          aria-label="Go to the Next Month"
          class="rdp-button_next"
          type="button"
        >
          <svg
            class="rdp-chevron"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <polygon
              points="8 18.612 14.1888889 12.5 8 6.37733333 9.91111111 4.5 18 12.5 9.91111111 20.5"
            />
          </svg>
        </button>
      </nav>
      <div
        class="rdp-month"
      >
        <div
          class="rdp-month_caption"
        >
          <span
            aria-live="polite"
            class="rdp-caption_label"
            role="status"
          >
            November 2021
          </span>
        </div>
        <table
          aria-label="November 2021"
          aria-multiselectable="false"
          class="rdp-month_grid"
          role="grid"
        >
          <thead
            aria-hidden="true"
          >
            <tr
              class="rdp-weekdays"
            >
              <th
                aria-label="Sunday"
                class="rdp-weekday"
                scope="col"
              >
                Su
              </th>
              <th
                aria-label="Monday"
                class="rdp-weekday"
                scope="col"
              >
                Mo
              </th>
              <th
                aria-label="Tuesday"
                class="rdp-weekday"
                scope="col"
              >
                Tu
              </th>
              <th
                aria-label="Wednesday"
                class="rdp-weekday"
                scope="col"
              >
                We
              </th>
              <th
                aria-label="Thursday"
                class="rdp-weekday"
                scope="col"
              >
                Th
              </th>
              <th
                aria-label="Friday"
                class="rdp-weekday"
                scope="col"
              >
                Fr
              </th>
              <th
                aria-label="Saturday"
                class="rdp-weekday"
                scope="col"
              >
                Sa
              </th>
            </tr>
          </thead>
          <tbody
            class="rdp-weeks"
          >
            <tr
              class="rdp-week"
            >
              <td
                class="rdp-day rdp-hidden rdp-outside"
                data-day="2021-10-31"
                data-hidden="true"
                data-month="2021-10"
                data-outside="true"
                role="gridcell"
              />
              <td
                class="rdp-day"
                data-day="2021-11-01"
                role="gridcell"
              >
                <button
                  aria-label="Monday, November 1st, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  1
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-02"
                role="gridcell"
              >
                <button
                  aria-label="Tuesday, November 2nd, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  2
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-03"
                role="gridcell"
              >
                <button
                  aria-label="Wednesday, November 3rd, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  3
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-04"
                role="gridcell"
              >
                <button
                  aria-label="Thursday, November 4th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  4
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-05"
                role="gridcell"
              >
                <button
                  aria-label="Friday, November 5th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  5
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-06"
                role="gridcell"
              >
                <button
                  aria-label="Saturday, November 6th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  6
                </button>
              </td>
            </tr>
            <tr
              class="rdp-week"
            >
              <td
                class="rdp-day"
                data-day="2021-11-07"
                role="gridcell"
              >
                <button
                  aria-label="Sunday, November 7th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  7
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-08"
                role="gridcell"
              >
                <button
                  aria-label="Monday, November 8th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  8
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-09"
                role="gridcell"
              >
                <button
                  aria-label="Tuesday, November 9th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  9
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-10"
                role="gridcell"
              >
                <button
                  aria-label="Wednesday, November 10th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  10
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-11"
                role="gridcell"
              >
                <button
                  aria-label="Thursday, November 11th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  11
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-12"
                role="gridcell"
              >
                <button
                  aria-label="Friday, November 12th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  12
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-13"
                role="gridcell"
              >
                <button
                  aria-label="Saturday, November 13th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  13
                </button>
              </td>
            </tr>
            <tr
              class="rdp-week"
            >
              <td
                class="rdp-day"
                data-day="2021-11-14"
                role="gridcell"
              >
                <button
                  aria-label="Sunday, November 14th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  14
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-15"
                role="gridcell"
              >
                <button
                  aria-label="Monday, November 15th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  15
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-16"
                role="gridcell"
              >
                <button
                  aria-label="Tuesday, November 16th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  16
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-17"
                role="gridcell"
              >
                <button
                  aria-label="Wednesday, November 17th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  17
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-18"
                role="gridcell"
              >
                <button
                  aria-label="Thursday, November 18th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  18
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-19"
                role="gridcell"
              >
                <button
                  aria-label="Friday, November 19th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  19
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-20"
                role="gridcell"
              >
                <button
                  aria-label="Saturday, November 20th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  20
                </button>
              </td>
            </tr>
            <tr
              class="rdp-week"
            >
              <td
                class="rdp-day"
                data-day="2021-11-21"
                role="gridcell"
              >
                <button
                  aria-label="Sunday, November 21st, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  21
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-22"
                role="gridcell"
              >
                <button
                  aria-label="Monday, November 22nd, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  22
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-23"
                role="gridcell"
              >
                <button
                  aria-label="Tuesday, November 23rd, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  23
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-24"
                role="gridcell"
              >
                <button
                  aria-label="Wednesday, November 24th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  24
                </button>
              </td>
              <td
                class="rdp-day rdp-today"
                data-day="2021-11-25"
                data-today="true"
                role="gridcell"
              >
                <button
                  aria-label="Today, Thursday, November 25th, 2021"
                  class="rdp-day_button"
                  tabindex="0"
                  type="button"
                >
                  25
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-26"
                role="gridcell"
              >
                <button
                  aria-label="Friday, November 26th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  26
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-27"
                role="gridcell"
              >
                <button
                  aria-label="Saturday, November 27th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  27
                </button>
              </td>
            </tr>
            <tr
              class="rdp-week"
            >
              <td
                class="rdp-day"
                data-day="2021-11-28"
                role="gridcell"
              >
                <button
                  aria-label="Sunday, November 28th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  28
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-29"
                role="gridcell"
              >
                <button
                  aria-label="Monday, November 29th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  29
                </button>
              </td>
              <td
                class="rdp-day"
                data-day="2021-11-30"
                role="gridcell"
              >
                <button
                  aria-label="Tuesday, November 30th, 2021"
                  class="rdp-day_button"
                  tabindex="-1"
                  type="button"
                >
                  30
                </button>
              </td>
              <td
                class="rdp-day rdp-hidden rdp-outside"
                data-day="2021-12-01"
                data-hidden="true"
                data-month="2021-12"
                data-outside="true"
                role="gridcell"
              />
              <td
                class="rdp-day rdp-hidden rdp-outside"
                data-day="2021-12-02"
                data-hidden="true"
                data-month="2021-12"
                data-outside="true"
                role="gridcell"
              />
              <td
                class="rdp-day rdp-hidden rdp-outside"
                data-day="2021-12-03"
                data-hidden="true"
                data-month="2021-12"
                data-outside="true"
                role="gridcell"
              />
              <td
                class="rdp-day rdp-hidden rdp-outside"
                data-day="2021-12-04"
                data-hidden="true"
                data-month="2021-12"
                data-outside="true"
                role="gridcell"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
